storage/innobase/btr/btr0cur.cc

```diff
@@ -1126,7 +1126,7 @@ static ulint btr_node_ptr_max_size(const dict_index_t* index)
 		/* Determine the maximum length of the index field. */
 
 		field_max_size = dict_col_get_fixed_size(col, comp);
-		if (field_max_size) {
+		if (field_max_size && field->fixed_len) {
```

Comment:
I remember that any `CHAR` that is longer than 255 bytes would in fact be stored in the same way as `VARCHAR`. If that is the case, then I think that the problem should be reproducible already with a 64-char prefix of a UTF-32 encoded column. Can you try to find out where exactly we would assign the `field->fixed_length`, and add a comment here that explains when we may have `field->fixed_len==0` for something that actually is a fixed-length field?

Reply:
In dict_index_add_col():
```
      /* Long fixed-length fields that need external storage are treated as
        variable-length fields, so that the extern flag can be embedded in
        the length word. */

        if (field->fixed_len > DICT_MAX_FIXED_COL_LEN) {
                field->fixed_len = 0;
        }
```
This was added in mysql-5.6, I guess. we need the prefix length to greater than 768 at least to repeat the scenario.